Source/Android/app/src/main/res/values/strings.xml

```diff
@@ -923,4 +923,8 @@ It can efficiently compress both junk data and encrypted Wii data.
     <string name="incompatible_figure_selected">Incompatible Figure Selected</string>
     <string name="select_compatible_figure">Please select a compatible figure file</string>
 
+    <string name="emulate_wii_speak">Wii Speak</string>
+    <string name="disconnect_wii_speak">Mute Wii Speak</string>
+    <string name="wii_speak_permission_warning">Missing RECORD_AUDIO permission</string>
+    <string name="wii_speak_permission_warning_description">Wii Speak emulation requires RECORD_AUDIO permission. Please enable this permission to allow your microphone use in-game. You might need to restart the game for the permission to be effective. </string>
```

Comment:
```suggestion
    <string name="wii_speak_permission_warning">Missing Microphone Permission</string>
    <string name="wii_speak_permission_warning_description">Wii Speak emulation requires microphone permission. Please enable this permission to allow your microphone use in-game. You might need to restart the game for the permission to be effective.</string>
```

Also, it would be nice UX to not mention having to restart if no game is running. (You can check this using NativeLibrary.IsRunning.)

Reply:
Please also note that displayAlertMsg uses a toast message if we're not currently in EmulationActivity. Toast messages are very limited in how much text they display on modern versions of Android, and for good reason – they only stick on screen for a short amount of time. If you want to display this message as a toast, it should be as short as possible. At the very least you should get rid of the "Please enable this permission to allow your microphone use in-game." sentence.

Reply:
The warning shouldn't be displayed when the game isn't running as the change is done by the settings switch and the activity is passed to the function. The C++ code should only be triggered when the emulation is running and when the Microphone class wants to start a stream, so I don't think using NativeLibrary.IsRunning is necessary here.

Otherwise, I shortened the message as you suggested.

Reply:
Sorry, I had missed the `activity == null` check in `requestRecordAudioPermission`. Okay, that sounds reasonable then. But consider expanding the comment in `requestRecordAudioPermission` to mention why we need the warning when calling from C++ specifically.

Reply:
Done.